pages/_app.js

```diff
@@ -36,6 +44,7 @@ export default function App({ Component, pageProps }) {
         activities={activities}
```

Comment:
You don't need this. Please don't duplicate your source of truth. Since you work with the 'activitiesInfo', which holds all needed information, you don't have to pass another activities array to the app. This will cause many bugs in the future.